var/spack/repos/builtin/packages/py-scipy/package.py

```diff
@@ -129,8 +145,37 @@ def setup_build_environment(self, env):
             if self.spec.satisfies("^py-numpy@1.16:1.17"):
                 env.set("NPY_DISTUTILS_APPEND_FLAGS", "1")
 
-        # Pick up Blas/Lapack from numpy
-        self.spec["py-numpy"].package.setup_build_environment(env)
+        # Pick up BLAS/LAPACK from numpy
+        if self.spec.satisfies("@:1.8"):
+            self.spec["py-numpy"].package.setup_build_environment(env)
+
+    # TODO: remove once pip build supports BLAS/LAPACK specification
+    # https://github.com/FFY00/meson-python/pull/122
+    @when("@1.9:")
+    def install(self, spec, prefix):
+        args = [
+            "setup",
+            "build",
+            "-Dblas=" + spec["blas"].libs.names[0],
```

Comment:
Is there a way to specify multiple libs to link to? Maybe `-Dblas=foo,bar` or `-Dblas=foo -Dblas=bar`? I think this is only needed for MKL.

Reply:
I don't think so at the moment. I'm not sure it's needed. As far as I can tell from `numpy/distutils/system_info.py`, the only library that is linked is `mkl_rt`. Doesn't `-Dblas=mkl -Dlapack=mkl` work?

Reply:
I try to avoid using MKL myself, but https://www.intel.com/content/www/us/en/developer/tools/oneapi/onemkl-link-line-advisor.html#gs.9hdj4v recommends using several libraries in the link line. We try to emulate this in Spack.

Reply:
Ah fun, fairly sure that we've never had a need for a link line like
```
-L${MKLROOT}/lib/ia32 -Wl,--no-as-needed -lmkl_gf -lmkl_intel_thread -lmkl_core -liomp5 -lpthread -lm -ldl
```
We can't expect that to come from the user though. This should come from `pkg-config`, from `cmake`, or (as a fallback) from a custom dependency method in Meson itself. We don't have the latter, so you should be able to use `=mkl` and you should get these multiple options from `pkg-config` or `cmake` right now already.

Reply:
Okay, guess we'll keep it as is and wait for the bug reports to trickle in. I suspect that the current level of customization with meson isn't sufficient for power users on HPC clusters, but maybe I'm just paranoid.

Reply:
That is perhaps true, since it didn't have much testing. But for MKL, the status quo should already be better than we had with `numpy.distutils`. Unless folks were manually hardcoding everything from the link line advisor in their own `site.cfg` file and never opening a bug report against numpy.

Reply:
> Unless folks were manually hardcoding everything from the link line advisor in their own `site.cfg` file and never opening a bug report against numpy.

For a second, I was about to say that's exactly what we do, but it looks like we have a special case for MKL where we always use `mkl_rt`: https://github.com/spack/spack/blob/v0.18.1/var/spack/repos/builtin/packages/py-numpy/package.py#L228,L248. So I'll do the same thing here.